Runtime/Scripts/RTCStats.cs

```diff
@@ -0,0 +1,809 @@
+                { RTCStatsType.OutboundRtp, ptr => new RTCOutboundRTPStreamStats(ptr) },
+                { RTCStatsType.RemoteInboundRtp, ptr => new RTCRemoteInboundRtpStreamStats(ptr) },
+                { RTCStatsType.RemoteOutboundRtp, ptr => new RTCRemoteOutboundRtpStreamStats(ptr) },
+                { RTCStatsType.MediaSource, ptr => new RTCMediaSourceStats(ptr) },
+                { RTCStatsType.Csrc, ptr => new RTCCodecStats(ptr) },
+                { RTCStatsType.PeerConnection, ptr => new RTCPeerConnectionStats(ptr) },
+                { RTCStatsType.DataChannel, ptr => new RTCDataChannelStats(ptr) },
+                { RTCStatsType.Stream, ptr => new RTCMediaStreamStats(ptr) },
+                { RTCStatsType.Track, ptr => new RTCMediaStreamTrackStats(ptr) },
+                { RTCStatsType.Transceiver, ptr => new RTCTransceiverStats(ptr) },
+                { RTCStatsType.Sender, ptr => new RTCSenderStats(ptr) },
+                { RTCStatsType.Receiver, ptr => new RTCReceiverStats(ptr) },
+                { RTCStatsType.Transport, ptr => new RTCTransportStats(ptr) },
+                { RTCStatsType.SctpTransport, ptr => new RTCTransportStats(ptr) },
+                { RTCStatsType.CandidatePair, ptr => new RTCIceCandidatePairStats(ptr) },
+                { RTCStatsType.LocalCandidate, ptr => new RTCIceCandidateStats(ptr) },
+                { RTCStatsType.RemoteCandidate, ptr => new RTCIceCandidateStats(ptr) },
+                { RTCStatsType.Certificate, ptr => new RTCCertificateStats(ptr) },
+            };
+        }
+
+        public static RTCStats Create(RTCStatsType type, IntPtr ptr)
+        {
+            return m_map[type](ptr);
+        }
+    }
+
+    public class RTCStatsReport : IReadOnlyDictionary<RTCStatsType, RTCStats>
+    {
+        private IntPtr self;
+        private readonly Dictionary<RTCStatsType, RTCStats> m_dictStats;
+
+        internal RTCStatsReport(IntPtr ptr)
+        {
+            self = ptr;
+            uint length = 0;
+            IntPtr ptrStatsTypeArray = IntPtr.Zero;
+            IntPtr ptrStatsArray = NativeMethods.StatsReportGetStatsList(self, ref length, ref ptrStatsTypeArray);
+
+            IntPtr[] array = ptrStatsArray.AsArray<IntPtr>((int)length);
+            byte[] types = ptrStatsTypeArray.AsArray<byte>((int)length);
+
+            m_dictStats = new Dictionary<RTCStatsType, RTCStats>();
+            for (int i = 0; i < length; i++)
+            {
+                RTCStatsType type = (RTCStatsType)types[i];
+                m_dictStats[type] = StatsFactory.Create(type, array[i]);
+            }
+        }
+
+        public bool ContainsKey(RTCStatsType key)
+        {
+            return m_dictStats.ContainsKey(key);
+        }
+
+        public bool TryGetValue(RTCStatsType key, out RTCStats value)
+        {
+            return m_dictStats.TryGetValue(key, out value);
+        }
+
+        public IEnumerable<RTCStatsType> Keys
```

Comment:
I think we will have less code if we can provide a function that returns `m_dictStats` as `IDictionary`, so that the caller can call `Keys`, `Values`, etc